test/rest_api/test_compaction_task.py

```diff
@@ -17,13 +18,15 @@ def check_compaction_task(cql, this_dc, rest_api, run_compaction, compaction_typ
                 cql.execute(stmt, [0, 'hello'])
                 cql.execute(stmt, [1, 'world'])
 
+                clear_module_tasks(rest_api, module_name)
```

Comment:
this call site seems rather late.
I think it would be better if placed in the beginning of the test.l while each call to `check_compaction_task(` should cleanup after itself and leave no compaction task behind.

Calling `drain_module_tasks` here might cover up cases where we forget to wait for tasks started here.